lib/ACL/ACLManager.php

```diff
@@ -98,13 +99,22 @@ private function getRules(array $paths, bool $cache = true): array {
 	 * @return string[]
 	 */
 	private function getRelevantPaths(string $path): array {
-		$paths = [$path];
+		$paths = [];
+		$fromTrashbin = str_starts_with($path, '__groupfolders/trash/');
+		if ($fromTrashbin) {
+			$rootName = explode('/', $path, 5)[3];
+			$rootName = substr($rootName, 0, strrpos($rootName, '.d'));
+		}
 		while ($path !== '') {
+			$paths[] = $path;
 			$path = dirname($path);
-			if ($path === '.' || $path === '/') {
+			if ($fromTrashbin && ($path === '__groupfolders/trash')) {
+				$trashItemRow = $this->trashManager->getTrashItemByFileName($rootName);
+				$path = dirname('__groupfolders/' . $trashItemRow['folder_id'] . '/' . $trashItemRow['original_location']);
+				$fromTrashbin = false;
+			} elseif ($path === '.' || $path === '/') {
```

Comment:
Maybe clearer, and a comment would no hurt.

```suggestion
			if ($fromTrashbin && ($path === '__groupfolders/trash')) {
				$trashItemRow = $this->trashManager->getTrashItemByFileName($rootName);
				$path = dirname('__groupfolders/' . $trashItemRow['folder_id'] . '/' . $trashItemRow['original_location']);
				$fromTrashbin = false;
				continue;
			}

			if ($path === '.' || $path === '/') {
```